### PR TITLE
Intel® PyTorch notebook container

### PR DESCRIPTION
**Pull request by sharvil10**
## Description

This PR adds containers for Intel® optimized TensorFlow with Intel® GPU and CPU support. There are 3 containers that are added.

The Base Image: This contains necessary driver level-zero installations for GPU support
Runtime Image: This image contains the pip packages for Intel Extension for PyTorch 2.1.110 and PyTorch 2.1.
Jupyter image: This image adds necessary jupyterhub packages on top of the runtime image

## How Has This Been Tested?

This was tested using makefile's deploy, tests and validate commands.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by atheo89** [comment]
Hello @sharvil10, thank you so much for your contribution to the opendatahub notebooks repository! 
I'll go ahead and move it to /ok-to-test to enable the CI tests. :slightly_smiling_face:

File: intel/runtimes/pytorch/ubi9-python-3.9/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    rm -f ./Pipfile.lock && \
+    # Fix permissions to support pip in Openshift environments \
+    chmod -R g+w /opt/app-root/lib/python3.9/site-packages && \
+    fix-permissions /opt/app-root -P
+
+ENV CPU_ENV=/opt/app-root-cpu
+COPY --chown=1001:0 Pipfile.lock.cpu ${CPU_ENV}/Pipfile.lock
+
+#CPU env
+RUN cd ${APP_ROOT2} && \
```
Comment (atheo89) [inline intel/runtimes/pytorch/ubi9-python-3.9/Dockerfile:47]:
```suggestion
RUN cd ${CPU_ENV} && \
```
Atherwise, this line fails to my end when I attend to run it locally with this error: `Could not open requirements file: [Errno 21] Is a directory: '/opt/app-root/requirements.txt'`

Comment (sharvil10) [reply intel/runtimes/pytorch/ubi9-python-3.9/Dockerfile]: Updated.

**Comment by sharvil10** [comment]
> Hello @sharvil10, thank you for your contribution to the opendatahub notebooks repository! I'll go ahead and move it to /ok-to-test to enable the CI tests. 🙂 Moreover, could you share with us the initiative of this contribution?

Hi @atheo89 thank you. I have fixed the issue with the build now. Could you please mark it `ok-to-test`  again? This is part of Intel's initiative to enable Intel® optimized AI frameworks on OpenShift. Could you please share what information do you need?

**Comment by harshad16** [comment]
/ok-to-test

Require some time next week to review this.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

the image build is successful for pytorch.

**Pull request merged**